### DISCUSSION TITLE
Gene name search using gff file not working

### DISCUSSION THREAD

**Discussion by FabianDK** [GMOD/jbrowse-components#3101] [Q&A]
Dear devs,

I am trying to implement the gene search function in a local instance of JBrowse2 using the CLI, but it does not work.
I am following the official instructions on the JBrowse2 page and then add the assembly and gff file:
```
# Install JBrowse2
npm install -g @jbrowse/cli
cd jbrowse
jbrowse create jbrowse2
npm install ansi-regex serve@14.0.1

# version
jbrowse --version
@jbrowse/cli/2.0.1 linux-x64 node-v18.6.0
```

To add assembly and gff:
```
# Create sorted gff file for JBrowse2
gt gff3 -sortlines -tidy -retainids GCF_000001635.27_GRCm39_genomic.gff > GCF_000001635.27_GRCm39_genomic.sorted.gff
bgzip GCF_000001635.27_GRCm39_genomic.sorted.gff
tabix GCF_000001635.27_GRCm39_genomic.sorted.gff.gz

# Add assembly and gff track
jbrowse add-assembly /jbrowse/genomes/GCF_000001635.27_GRCm39_genomic.fna --out /jbrowse/jbrowse2 --load copy --alias GRCm39
jbrowse add-track /jbrowse/genomes/GCF_000001635.27_GRCm39_genomic.sorted.gff.gz --out /jbrowse/jbrowse2 --assemblyNames GRCm39 --load copy

# Index tracks to enable gene name searching
jbrowse text-index --out /jbrowse/jbrowse2 -a GRCm39
```
Finally, I run the npx serve command and can connect through the browser. The assembly and gff track can be loaded and visualized as expected. Also the feature details of each gene are correctly shown. If I however type a gene name or geneID into the search field (e.g. the gene Tex44), I get: Error: Unknown reference sequence "Tex44"

I am running JBrowse2 in a docker container with the base image node:18. I thought that after running the text-index function, the gene names would be searchable, but this is not the case.
I can confirm that in the jbrowse2/trix folder, the GRCm39.ix, GRCm39.ixx and GRCm39_meta.json are present.

I would really appreciate some help solving this.

**Comment by cmdcolin** [comment]
Hi there
This is an interesting one, thanks for reporting.

Basically, if you change you add-assembly command to use --name GRCm39 instead of --alias GRCm39 it should work

We could possibly consider fixing this as a bug, but hopefully that helps!

**Reply by cmdcolin** [reply]
(you can probably also edit your config.json after the fact to change the assembly "name" field to GRCm39)

**Comment by FabianDK** [comment]
Many thanks. Changing --alias to --name fixed it!
